scripts/check-fixed-packages-ui-extensions.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+const getReleasePlan = require('@changesets/get-release-plan').default;
+
+const packageInfo = require('./package-info');
+
+/**
+ * Checks that if any `@datadog/ui-extensions-*` package is getting a release,
+ * all `@datadog/ui-extensions-*` packages are getting a release.
+ *
+ * While `changesets` does have support for "linked" packages:
+ * https://github.com/changesets/changesets/blob/main/docs/linked-packages.md,
+ * it's not exactly what we want.
+ * The way "linked" packages work,
+ * we can still leave out packages that we want to be on the same version.
+ *
+ * We do still want "linked" packages in our `.changeset/config.json`,
+ * since it means we'll always get everything on the same version.
+ * To that end,
+ * this check doesn't need to actually look at specific version numbers.
+ * We just have to catch packages that might be missed.
+ *
+ * The behavior we want is actually what's being proposed as "fixed" packages:
+ * https://github.com/changesets/changesets/pull/690.
+ * Until that feature is released,
+ * we have this check to make sure we release packages in lock-step.
```

Comment:
wow thank you for enforcing this in code and documenting so well. I tend to opt for manual enforcement of such things, which inevitably ends up not working